include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -15,35 +15,78 @@
 
 #include <limits>
 
+#include <seqan3/alignment/matrix/detail/coordinate_matrix.hpp>
 #include <seqan3/alignment/matrix/detail/matrix_coordinate.hpp>
 #include <seqan3/alignment/pairwise/detail/type_traits.hpp>
 #include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/concept/tuple.hpp>
 #include <seqan3/core/type_traits/template_inspection.hpp>
 
 namespace seqan3::detail
 {
 
+/*!\brief A function object that compares and possibly updates the alignment optimum with the current cell.
+ * \ingroup pairwise_alignment
+ *
+ * \details
+ *
+ * Updates the current alignment optimum with the new score and the respective coordinate if the new score
+ * compares greater or equal to the score of the current optimum.
+ */
+struct max_score_updater
+{
+    /*!\brief Compares and updates the optimal score-coordinate pair.
+     * \tparam score_t The type of the score to track; must model std::totally_ordered and
+     *                 std::assignable_from `const & score_t`.
+     * \tparam coordinate_t The type of the coordinate to track; must model std::assignable_from `const & coordinate_t`.
+     *
+     * \param[in,out] optimal_score The optimal score to update.
+     * \param[in,out] optimal_coordinate The optimal to update.
+     * \param[in] current_score The score of the current cell.
+     * \param[in] current_coordinate The coordinate of the current cell.
+     *
+     * \details
+     *
+     * Compares the current_score with the optimal score and updates the optimal score and coordinate if the current
+     * one is the new optimum. Otherwise, keeps the old optimum.
+     */
+    template <typename score_t, typename coordinate_t>
+    //!\cond
+        requires (std::totally_ordered<score_t> && std::assignable_from<score_t &, score_t const &> &&
+                  std::assignable_from<coordinate_t &, coordinate_t const &>)
+    //!\endcond
+    void operator()(score_t & optimal_score,
+                    coordinate_t & optimal_coordinate,
+                    score_t const & current_score,
+                    coordinate_t const & current_coordinate) const noexcept
+    {
+        bool const is_better_score = current_score >= optimal_score;
+        optimal_score = (is_better_score) ? current_score : optimal_score;
+        optimal_coordinate = (is_better_score) ? std::move(current_coordinate) : optimal_coordinate;
+    }
+};
+
 /*!\brief Implements the tracker to store the global optimum for a particular alignment computation.
  * \ingroup pairwise_alignment
  *
  * \tparam alignment_configuration_t The type of the alignment configuration; must be a type specialisation of
  *                                   seqan3::configuration.
- *
+ * \tparam binary_updater_t The type of a binary update operation to update the alignment optimum; must model
+ *                          std::semiregular and must be invocable with two tuples each containing the score and
+ *                          the coordinate.
```

Comment:
Isn't your `max_score_updater` one example for this type? If so `max_score_updater` does not get two tuples?

Reply:
Ah I need to update this documentation. Thanks for pointing this out.